app/components/Portfolio/Asset.tsx

```diff
@@ -0,0 +1,137 @@
+import React from "react";
+import { AssetData } from "@/app/store/slices/portfolioSlice";
+import Image from "next/image";
+import * as Icons from "@/app/icons";
+import ProgressBar from "../ProgressBar";
+
+export interface AssetProps {
+  asset: AssetData;
+  onEdit: () => void;
+}
+
+const Asset: React.FC<AssetProps> = ({ asset, onEdit }) => {
+  return (
+    <div className="flex border bg-white dark:bg-[#191925] dark:border-[#19192e] rounded-lg mb-6">
+      <div className="flex-1 p-2 bg-[#a2a3e8] dark:bg-[#191932] rounded-l-lg">
+        <div className="p-2">
+          <div className="flex justify-between mt-2">
+            <div className="flex space-x-2">
+              <img src={asset.image} className="w-7" alt={asset.name} />
+              <span className="text-[#3c3c7e] dark:text-white text-2xl">
+                {asset.name} ({asset.symbol?.toUpperCase()})
+              </span>
+            </div>
+            <button className="bg-[#3c3c7e] rounded-md p-1" onClick={onEdit}>
+              <Image className="w-6" src={Icons.Edit} alt="edit" />
+            </button>
+          </div>
+          <div className="dark:text-secondary mt-8">Total Value</div>
+          <div className="flex space-x-2 items-center mt-1">
+            <span className="text-[#3c3c7e] dark:text-white text-2xl">
+              ${asset.total_value.toLocaleString()} USD
+            </span>
+            {asset.profit_percentage > 0 ? (
+              <Image className="w-5" src={Icons.UpArrow} alt="+" />
+            ) : (
+              <Image className="w-5" src={Icons.DownArrow} alt="-" />
+            )}
+            <span
+              className={
+                asset.profit_percentage > 0
+                  ? "text-[#00b1a6]"
+                  : "text-[#fe2264]"
+              }
+            >
+              {asset.profit_percentage}%
+            </span>
```

Comment:
you should make this a component

Reply:
I've made a reusable component for it <PercentageChange />